shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -427,12 +429,80 @@ - (void)onPreEngineRestart {
   [self initializeKeyboard];
 }
 
+- (void)notifySemanticsEnabledChanged {
+  BOOL mySemanticsEnabled = !!_bridge;
+  BOOL newSemanticsEnabled = _engine.semanticsEnabled;
```

Comment:
Consider making the engine pass its `semanticsEnabled` value as parameter to this method. This would align better with the corresponding embedder API. I don't feel strongly about this, feel free to skip if you prefer as is

Reply:
Yeah I prefer the current way. Since `FVC` already references the engine anyway, the current way follows SSOT, so that `FVC` doesn't have to worry about this argument being different from its `engine.semanticsEnabled`.